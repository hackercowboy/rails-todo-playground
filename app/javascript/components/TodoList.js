import React from 'react';
import PropTypes from 'prop-types';
import TodoListItem from './TodoListItem';

const e = React.createElement;

function TodoList({ todos, onDelete, onDone }) {
  return e(
    'ul',
    { className: 'todo-list' },
    todos.map((todo) => e(TodoListItem, {
      ...todo, key: todo.id, onDelete, onDone,
    })),
  );
}

TodoList.propTypes = {
  todos: PropTypes.arrayOf(PropTypes.shape({ id: PropTypes.number.isRequired })).isRequired,
  onDelete: PropTypes.func.isRequired,
  onDone: PropTypes.func.isRequired,
};

export default TodoList;
