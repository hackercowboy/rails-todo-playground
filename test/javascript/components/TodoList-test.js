import React from 'react';
import { shallow } from 'enzyme';

import TodoList from '../../../app/javascript/components/TodoList';

const e = React.createElement;

describe('<TodoList/>', () => {
  it('should render', () => {
    const wrapper = shallow(e(TodoList, {
      todos: [{ id: 32, text: 'Test', status: 'pending' }],
      onDone: jest.fn(),
      onDelete: jest.fn(),
    }));
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
