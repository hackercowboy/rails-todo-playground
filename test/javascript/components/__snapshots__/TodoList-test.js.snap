// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList/> should render 1`] = `
"<ul className=\\"todo-list\\">
  <TodoListItem id={32} text=\\"Test\\" status=\\"pending\\" onDelete={[Function: mockConstructor] { _isMockFunction: true, getMockImplementation: [Function (anonymous)], mock: Object [Object: null prototype] { calls: [], instances: [], invocationCallOrder: [], results: [] }, mockClear: [Function (anonymous)], mockReset: [Function (anonymous)], mockRestore: [Function (anonymous)], mockReturnValueOnce: [Function (anonymous)], mockResolvedValueOnce: [Function (anonymous)], mockRejectedValueOnce: [Function (anonymous)], mockReturnValue: [Function (anonymous)], mockResolvedValue: [Function (anonymous)], mockRejectedValue: [Function (anonymous)], mockImplementationOnce: [Function (anonymous)], mockImplementation: [Function (anonymous)], mockReturnThis: [Function (anonymous)], mockName: [Function (anonymous)], getMockName: [Function (anonymous)] }} onDone={[Function: mockConstructor] { _isMockFunction: true, getMockImplementation: [Function (anonymous)], mock: Object [Object: null prototype] { calls: [], instances: [], invocationCallOrder: [], results: [] }, mockClear: [Function (anonymous)], mockReset: [Function (anonymous)], mockRestore: [Function (anonymous)], mockReturnValueOnce: [Function (anonymous)], mockResolvedValueOnce: [Function (anonymous)], mockRejectedValueOnce: [Function (anonymous)], mockReturnValue: [Function (anonymous)], mockResolvedValue: [Function (anonymous)], mockRejectedValue: [Function (anonymous)], mockImplementationOnce: [Function (anonymous)], mockImplementation: [Function (anonymous)], mockReturnThis: [Function (anonymous)], mockName: [Function (anonymous)], getMockName: [Function (anonymous)] }} />
</ul>"
`;
